source/common/http/conn_manager_impl.cc

```diff
@@ -1575,16 +1669,20 @@ void ConnectionManagerImpl::ActiveStreamFilterBase::commonContinue() {
   if (trailers()) {
     doTrailers();
   }
+
+  if (stoppedAll()) {
+    allowIteration();
```

Comment:
commonHandleAfterDataCallback() does not set iteration_state_ to Continue if iteration_state_ was StopAllIteration. That's because decodeTrailers() (triggered by doTrailers) needs StopAllIteration to signal operating on the current filter rather than the next.

In decodeTrailers(), we set iteration_state_ to Continue because decodeTrailers() may call addDecodedData() and then decodeData() which should operate on the next filter rather than the current. 

So this piece of code will be called when the filter has been stopped all and the stream has no trailers.

New comments added. Thanks!

Reply:
I suspect it may be faster to sync in person but I'm failing to see how we skip setting iteration_state_  when processing data if all iteration was previously stopped and we just resumed.  If we do, I'm wondering how we handle the case where the headers stopped all iteration, we processed onData and then differentiate onData returning StopIteration from Continue.

Reply:
There indeed is a problem! Thanks for catching this!!

I could have missed other bugs you pointed out. But I got the point that we can't differentiate StopIteration from StopAllIteration for the following case: if the StopAll filter's decodeData() returns Continue, iteration_state_ will not be resumed in commonHandleAfterDataCallback(). But if the decodeData() returns StopIteration, iteration_state_ will be set to IterationState::StopSingleIteration, and decodeTrailers() will not iterate from the current filter. 

I added a bool iterate_from_current_filter_ to indicate if the iteration should start with the current filter. And resume iteration_state_ to continue before call doData() and doTrailers() in commonContinue().